Swerve additions

KeremAlp01 opened the pull request:
alliance,vendor,state and SwerveModule

emrecandabak committed:
Fix formatting

KeremAlp01 committed:
Needed Things and SwerveModule

alliance,vendor,state and SwerveModule

KeremAlp01 committed:
Add swervemodule and vendors

alliance,vendor,state and SwerveModule

emrecandabak committed:
Update spotless version to latest

emrecandabak committed:
Update build.gradle

emrecandabak committed:
Merge branch 'main' into main

KeremAlp01 committed:
Merge branch 'main' of https://github.com/KeremAlp01/2023-offseason-rewrite

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+
+    public SwerveModule(int moduleNumber,SwerveModuleConstants moduleconstans){
+        this.moduleNumber = moduleNumber;
+        angleOffsett = moduleconstans.angleOffsett;
```
fix typo -> angleOffset

KeremAlp01 replied:
efsanevi ingilizcem

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+        lastangle = getState().angle;
+    }
+
+    private void configangleEncoder() {
```
camelCase

KeremAlp01 replied:
configAngleEncoder?

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+        driveEncoder= drivemotor.getEncoder();
+        driveController = drivemotor.getPIDController();
+        //config
+        anglemotor = new CANSparkMax(moduleconstans.angleMotorID,MotorType.kBrushless);
```
camelCase

KeremAlp01 replied:
driveMotor?

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+        angleEncoder = new CANCoder(moduleconstans.canCoderID);
+        configangleEncoder();
+        //config
+        drivemotor = new CANSparkMax(moduleconstans.driveMotorID,MotorType.kBrushless);
```
camelCase

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+    private void configangleEncoder() {
+        angleEncoder.configFactoryDefault();
+        CANCoderUtil.setCANCoderBusUsage(angleEncoder,CCUsage.kMinimal);
+        angleEncoder.configAllSettings(null);
```
??

KeremAlp01 replied:
nullu doldurmayı unutmuşum ayriyetten bir şey var mı

emrecandabak replied:
> nullu doldurmayı unutmuşum ayriyetten bir şey var mı

yok

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+
+    public void resetToAbsolute(){    
+        double absolutePosition = getAngle().getDegrees() - angleOffsett.getDegrees();
+        integradetAngleEncoder.setPosition(absolutePosition);
```
integratedAngleEncoder

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+    public void configDriveMotor(){
+        drivemotor.restoreFactoryDefaults();
+        CANSparkMaxUtil.setCANSparkMaxBusUsage(drivemotor, Usage.kAll);
+        drivemotor.getInverted();
```
getInverted?

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+        drivemotor.restoreFactoryDefaults();
+        CANSparkMaxUtil.setCANSparkMaxBusUsage(drivemotor, Usage.kAll);
+        drivemotor.getInverted();
+        drivemotor.getIdleMode();
```
getIdleMode?

KeremAlp01 replied:
neutral mode eklemeyecek miyim?

emrecandabak replied:
> neutral mode eklemeyecek miyim?

setlemen lazım o zaman

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+        drivemotor.setSmartCurrentLimit(60);
+        driveEncoder.getVelocityConversionFactor();
+        driveEncoder.getPositionConversionFactor();
+        driveController.getP();
```
get her şey???

KeremAlp01 replied:
bunlarda setmiş

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+        driveController.getD();
+        driveController.getFF();
+        drivemotor.burnFlash();
+        drivemotor.enableVoltageCompensation(12);
```
burnFlash en son yapılmalı her zaman

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+
+    private void setSpeed(SwerveModuleState desiredState,boolean isOpenLoop){
+        if(isOpenLoop){
+            double percentoutput = desiredState.speedMetersPerSecond / 4.5;
```
4.5 elle yazılmayacak,  constant olmalı

KeremAlp01 replied:
tamamdır constantları en son ayarlicam hata veriyordu 4.5 ondan yazdım 0.01 de constant

emrecandabak commented on src/SwerveModule.java:
```diff
@@ -0,0 +1,142 @@
+        }
+    }
+    private void setAngle(SwerveModuleState desiredState){
+        Rotation2d angle = (Math.abs(desiredState.speedMetersPerSecond) <= (4.5 * 0.01)) ? getAngle() :desiredState.angle;
```
4.5 elle yazılmayacak

KeremAlp01 committed:
Add utils and swerveSubsystem

utils,config,state,module fix and add swerve subsystem

emrecandabak committed:
Merge branch 'main' into main

KeremAlp01 committed:
Merge branch 'main' of https://github.com/KeremAlp01/2023-offseason-rewrite

KeremAlp01 committed:
Add utils and swerveSubsystem

utils,config,state,module fix and add swerve subsystem

emrecandabak commented on src/main/java/frc/lib/util/CTREConfigs.java:
```diff
@@ -1,8 +1,22 @@
+import frc.robot.Constants;
+
+public final class CTREConfigs {
+  public CANCoderConfiguration swerveCanCoderConfig;
```
swerveCANCoderConfig

emrecandabak commented on src/main/java/frc/robot/SwerveModule.java:
```diff
@@ -42,7 +44,7 @@ public SwerveModule(int moduleNumber,SwerveModuleConstants moduleconstans){
 
         //config
-        angleEncoder = new CANCoder(moduleconstans.canCoderID);
+        angleEncoder = new CANCoder(moduleconstans.cancoderID);
```
moduleConstants

emrecandabak replied:
CANCoderID

emrecandabak commented on src/main/java/frc/robot/SwerveModule.java:
```diff
@@ -77,8 +79,8 @@ public void resetToAbsolute(){
-        driveMotor.setInverted();
-        driveMotor.setIdleMode();
+        driveMotor.setInverted(false);
+        driveMotor.setIdleMode(null);
```
null?

KeremAlp01 replied:
hata veriyo yarın constant ekleyecem içine

emrecandabak commented on src/main/java/frc/robot/SwerveModule.java:
```diff
@@ -77,8 +79,8 @@ public void resetToAbsolute(){
+        driveMotor.setInverted(false);
+        driveMotor.setIdleMode(null);
         driveMotor.setSmartCurrentLimit(60);
         driveEncoder.getVelocityConversionFactor();
```
getler duruyor hala??

emrecandabak commented on src/main/java/frc/robot/SwerveModule.java:
```diff
@@ -94,8 +96,8 @@ public void configDriveMotor(){
+        angleMotor.setIdleMode(null);
         angleMotor.setSmartCurrentLimit(20);
         integradetAngleEncoder.setPositionConversionFactor(28);
         angleController.getP();
```
pid değerlerini "getlemek" ne işine yarayacak tam olarak? konfigüre ederken setlemen lazım

emrecandabak commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -99,29 +108,82 @@ public Swerve() {
+       fieldRelative ? 
+       ChassisSpeeds.fromFieldRelativeSpeeds(translation.getX(),translation.getY(), rotation ,  getPose().getRotation()) 
+       :new ChassisSpeeds(translation.getX(), translation.getY(), rotation));
+       SwerveDriveKinematics.desaturateWheelSpeeds(swerveModuleState, 4.5);
```
4.5 yazmıyorsun elinle

emrecandabak commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -99,29 +108,82 @@ public Swerve() {
+ 
+ 
+  public void setModuleStates(SwerveModuleState[] desiredStates){
+    SwerveDriveKinematics.desaturateWheelSpeeds(desiredStates, 4.5);
```
4.5

emrecandabak commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -99,29 +108,82 @@ public Swerve() {
+  public void resetOdometry(Pose2d pose){
+    estimator.resetPosition(getHeadingRotation(), getModulePositions(), pose);
+  }
+  public void EstimatorWithVision(Pose2d measurement,double latency){
```
fonksiyon ismini düzelt

emrecandabak commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -99,29 +108,82 @@ public Swerve() {
+    ;
+  }
+  public Rotation2d getYaw(){
+    return(true)?
```
bu ne?

KeremAlp01 replied:
bende gözükmüyor bu

KeremAlp01 replied:
invert gyro yerine yazdım constant ekledim mi hepsini değiştircem

emrecandabak commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -99,29 +108,82 @@ public Swerve() {
+  public boolean getIsFieldOriented(){
+    return isFieldOriented;
+  }
+  public void setisFieldOrieted(){
```
setFieldOriented*

emrecandabak commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -99,29 +108,82 @@ public Swerve() {
+  public void setisFieldOrieted(){
+    isFieldOriented = true;
+  }
+  public void isRobotOriented(){
```
setRobotOriented

emrecandabak reviewed (changes requested):
spotless ve pre-commit çalıştır ayrıca

emrecandabak commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -0,0 +1,189 @@
+
+        estimator =
+            new SwerveDrivePoseEstimator(
+                null,
```
null?

emrecandabak commented on src/main/java/frc/robot/SwerveModule.java:
```diff
@@ -0,0 +1,145 @@
+        driveController.getD();
+        driveController.getFF();
+        driveMotor.enableVoltageCompensation(12);
+        driveEncoder.setPosition(0.0);
```
burnFlashtan sonra sıfırlaman lazım encoderi

KeremAlp01 committed:
Fix swerve modules and subsystems

fix config,constants,subsystem,module

emrecandabak commented on src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -180,10 +180,10 @@ public Rotation2d getYaw(){
     return isFieldOriented;
   }
-  public void setisFieldOrieted(){
+  public void setFieldOrieted(){
```
oriented

emrecandabak commented on src/main/java/frc/robot/SwerveModule.java:
```diff
@@ -100,10 +99,10 @@ public void configAngleMotor(){
         angleMotor.setIdleMode(null);
         angleMotor.setSmartCurrentLimit(20);
         integradetAngleEncoder.setPositionConversionFactor(28);
```
integrated

BABAANNEN committed:
add LED animation

add multiple functions to make sliding led animation possible (might have bugs)

BABAANNEN committed:
Applied spotless

applied gradlew spotless apply

KeremAlp01 committed:
Add swerve commands and fixes

KeremAlp01 committed:
add some constants

KeremAlp01 committed:
fix config

BABAANNEN committed:
Enhanced next index command

now it turns back once it reaches the end of one side

BABAANNEN committed:
Add enum

BABAANNEN committed:
somehow added new animation

idk how

KeremAlp01 committed:
Add limiter constants

Batucagan committed:
Gradlew Update

i have 0 idea what changed

emrecandabak committed:
Kerem Alp Additions from my pc

BABAANNEN committed:
add enum and add blinking led effect

KeremAlp01 committed:
Add IntakeSubsystem

KeremAlp01 committed:
Add Constants

KeremAlp01 committed:
Fix subsystem

KeremAlp01 committed:
Add intakeStop

emrecandabak commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -10,56 +10,56 @@
+
 public class IntakeSubsystem extends SubsystemBase {
     private final WPI_TalonFX intakeMotor = new WPI_TalonFX(Constants.IntakeConstants.intakeMotorID);
     private StatorCurrentLimitConfiguration m_limit = new StatorCurrentLimitConfiguration();
```
Stator değil, supply limit yapmamız lazım. Konu hakkında detaylı bilgi: https://www.chiefdelphi.com/t/current-limiting-talonfx-values/374780

emrecandabak commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -10,56 +10,56 @@
         intakeMotor.configFactoryDefault();
         intakeMotor.setNeutralMode(Constants.IntakeConstants.motorNeutralMode);
         intakeMotor.setInverted(Constants.IntakeConstants.intakeMotorInverted);
         intakeMotor.configOpenloopRamp(0);
```
OpenloopRamp constant olmalı. https://v5.docs.ctr-electronics.com/en/stable/ch13_MC.html#ramping ne olduğu ile ilgili bilgi

emrecandabak commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -10,56 +10,56 @@
         intakeMotor.configOpenloopRamp(0);
-        // config open loop???
+        //config open loop???
         m_limit.triggerThresholdCurrent = 0;
```
Bu değerler de constant olmalı, enable hariç

emrecandabak commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -10,56 +10,56 @@
-
-    public void disableCurrentLimit() {
+    public void disableCurrentLimit(){
         intakeMotor.configStatorCurrentLimit(new StatorCurrentLimitConfiguration());
```
burası da değişeccek

emrecandabak reviewed (changes requested):
Intake çalıştırma komutu, çekilen akıma göre obje olup olmadığını algılama, obje varsa tutma eklenmeli

emrecandabak committed:
Merge branch 'main' into main

KeremAlp01 committed:
Fix subsystemv2

BABAANNEN committed:
blinking led animation sent in for review

KeremAlp01 committed:
Add IntakeStart

KeremAlp01 committed:
Fix IntakeSubsystem

KeremAlp01 committed:
Add HeadingSwerve

emrecandabak commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -31,7 +33,10 @@ public RobotContainer() {
 
-    private void configureBindings() {}
+    private void configureBindings() {
+        new JoystickButton(driver, 0).onTrue(new InstantCommand(() -> {m_swerve.setWantsHeadingLock(true);}))
```
0. buton diye bir şey yok

Batucagan committed:
nothing

Batucagan committed:
Fix issues found in LED animation review

it works now

Batucagan committed:
fix led buffer length

change buffer length from 60 to 18

emrecandabak committed:
Keremalp additions

emrecandabak committed:
Sync fork

emrecandabak committed:
Merge pull request #2 from Batucagan/main

Merge to fix error
